core/trino-main/src/test/java/io/trino/sql/planner/TestArraySortAfterArrayDistinctRewriter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.planner;
+
+import io.trino.sql.ExpressionTestUtils;
+import io.trino.sql.planner.assertions.PlanMatchPattern;
+import io.trino.sql.planner.iterative.rule.ArraySortAfterArrayDistinct;
+import io.trino.sql.planner.iterative.rule.test.BaseRuleTest;
+import io.trino.sql.planner.iterative.rule.test.PlanBuilder;
+import io.trino.sql.planner.plan.Assignments;
+import io.trino.sql.tree.Expression;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+
+import static io.trino.sql.planner.assertions.PlanMatchPattern.project;
+import static io.trino.sql.planner.assertions.PlanMatchPattern.values;
+
+public class TestArraySortAfterArrayDistinctRewriter
+        extends BaseRuleTest
+{
+    @Test
+    public void basic()
+    {
+        test("ARRAY_DISTINCT(ARRAY_SORT(ARRAY['a']))", "ARRAY_SORT(ARRAY_DISTINCT(ARRAY['a']))");
+    }
+
+//    @Test
+//    // TODO: should figure out why this is failing, although the plan seems correct
```

Comment:
Apparently, the assertions cannot handle the lambda. 

Please add this test case to `query`. Also, add some more examples there with unsorted and non-distinct arrays to show thet the results are correct.

Reply:
Which test you actually mean with `query`?

Reply:
Thanks, I actually figured out that one already :)